docs/rfcs/0001-video-retranscoding.md

```diff
@@ -0,0 +1,43 @@
+## Video Re-transcoding
+
+#### Abstract
+Provide a way to trigger re-transcoding of specified videos, or all videos in a collection, if new presets are needed.
+
+
+#### Architecture Changes
+
+- Create a flag on Collection and Video: boolean field ‘schedule_retranscode’
+- Create a recurring task to look for videos and collections with `schedule_retranscode=True`.
+  - check flagged collections first:
+     - set `schedule_retranscode=True` for every Video in the Collection, then
+     - set `schedule_retranscode=False` on the Collection
+  - Trigger a new encode job for each flagged Video.
+  - Assign a new video status - ‘Retranscoding’
+  - Set `schedule_retranscode=False` on the Video
+- Modify the generation of encoding job metadata:
+  - a temporary prefix should be added to the output filenames to avoid transcode errors that occur when trying to overwrite existing output from the previous transcode
+- Videos being re-transcoded should still be playable (front end needs to accept the new reencoding status as playable)
+- Modify the task that checks on video transcode statuses
+  - should handle the new Retranscoding status
+  - There will now be 2+ encode jobs for the video instead of just 1, the most recent is the one that needs to be checked for progress.
+- When a retranscoding is complete:
+  - copy output from the temporary S3 subfolder to the correct subfolder, then delete the temporary files
+  - Change video status back to complete
+- If a retranscoding task fails:
+  - Log the error
+  - Set video status back to complete (original transcode should still work)
+
+#### Security/Other Considerations
+
+If for some reason the copying of retranscoded S3 files is interrupted or fails partway, the video may be rendered unplayable, since some of the original transcode files will be overwritten.
+
```

Comment:
Can we turn on versioning for this bucket? I know it's expensive, but it's an easy way to get. backup.

Reply:
Not sure where logging belongs, but I'd like to have a way to follow along with the retrancoding status

Reply:
>Can we turn on versioning for this bucket?

@shaidar @blarghmatey that bucket would be the value of `VIDEO_S3_TRANSCODE_BUCKET`

Reply:
@pdpinch another option is to always rename the transcoded output files instead of overwriting the original ones, but that would mean updating the video links in xpro too.